src/IonParserBinaryRaw.ts

```diff
@@ -209,15 +209,22 @@ export class ParserBinaryRaw {
         return LongInt.fromVarIntBytes(bytes, isNegative);
     }
 
-    private readLongInt() : LongInt {
-        if (this._len - this._in.position() < 1) return new LongInt(0);
-        let bytes : Uint8Array = this._in.view(this._in.position(), );
-        let sign = (bytes[0] & 0x10) != 0;
-        bytes[0] &= 0x7F;
-        return LongInt.fromIntBytes(Array.prototype.slice.call(bytes), sign);
+    static readSignedIntFrom(input: BinarySpan, numberOfBytes: number) : LongInt {
```

Comment:
I've renamed this method from `readLongInt` to `readSignedInt` because the reader currently does distinguish between `Int`s that could fit in a standard javascript `Number` value and those which require a `BigInt` (`LongInt` in this library). All `Int` values (which only occur in `Timestamp`s and `Decimal` values) are read into `LongInt`s. Also, the method's signature indicates that it always returns a `LongInt`. 

I've also extracted the meat of this method out into a `static` method that is more easily unit tested.

Reply:
Perfect, yes were only going support ints over 32 bits in decimals and timestamps. Eventually we could build a number tower but it would require altering the api contracts.